src/__tests__/lib/rules/prefer-in-document.js

```diff
@@ -35,6 +35,11 @@ const valid = [
   ]),
   `expect(screen.notAQuery('foo-bar')).toHaveLength(1)`,
   `expect(screen.getByText('foo-bar')).toHaveLength(2)`,
+
+  `expect(foo).toBeDefined();`,
+  `expect(foo).not.toBeDefined();`,
+  `expect(foo).toBeNull();`,
+  `expect(foo).not.toBeNull();`,
```

Comment:
i think it's possible to get a reference to the declaration of a variable.  if so then we can check for the following as invalid: 

```js
const foo = screen.getByText("foo");
expect(foo).toBeDefined();

```

here's an example: https://astexplorer.net/#/gist/86b8b0096fdc670748c705c638569ab1/c05f3e8c258426cb505b10b139e75e2895edc253

Reply:
Yeah, I've played around with this a bit. I just wasn't sure on were to draw the line. Perhaps one reference check "backwards" to look for a clear reference to one of the queries is good enough. I see some scenarios where the variable passed to `expect` is some kind of array returned from a map or something.

Reply:
I've updated the logic a bit to check the variable definition. Should cover a lot of the usecases I've seen atleast 😊